### PR TITLE
Fix/enrollment form UI

### PR DESCRIPTION
**Pull request by Kokila-Kaavian**

### CODE REVIEW

File: app/dteconsentappclient/webapp/utils/ConsentAddressSuggestion.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+     * Get the input value while suggest
+     * @param {String} sValue user input
+     * @param {Object} oConsentModel consent model
+     * @param {String} sDTEAddressValidationUrl address validation url
+     */
+    onConsentAddrSuggestion: function(sValue, oConsentModel, sDTEAddressValidationUrl){
+      if (sValue.length > 3) {
+        // Trigger API call after 3 characters
+       this.fetchAddressSuggestions(sValue, oConsentModel, sDTEAddressValidationUrl);
+     }
```
Comment (Divya-Kaavian) [inline app/dteconsentappclient/webapp/utils/ConsentAddressSuggestion.js:16]:
```suggestion
      // Trigger API call after 3 characters
      if (sValue.length > 3)
        this.fetchAddressSuggestions(sValue, oConsentModel, sDTEAddressValidationUrl);
```

**Review by Divya-Kaavian — CHANGES_REQUESTED** [review]

File: app/dteconsentappclient/webapp/controller/ConsentForm.controller.js [hunk 2]
```diff
@@ -91,33 +103,65 @@ sap.ui.define([
 					});
 			},
 
+			// Get the value while suggest
+			onConsentAddrSuggest: function(oEvent){
+				// Get the entered value from input
+				const sValue = oEvent.getParameter("suggestValue");
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/ConsentForm.controller.js:109]: Changed

File: app/dteconsentappclient/webapp/controller/ConsentForm.controller.js [hunk 3]
```diff
@@ -91,33 +103,65 @@ sap.ui.define([
...
+				// To checks the condition and call the DTE address End point to get the suggestion list
+				ConsentAddressSuggestion.onConsentAddrSuggestion(sValue, oConsentModel, this.DTEAddressValidationUrl);
+			},
+
+			// To get the selected item from the suggestion list
+			onConsentAddrSugSelected: function(oEvent){
+			// Get the bound model
+			const oConsentModel = this.getView().getModel("oConsentModel");
+			
+			// To get the selected item from the suggestion list and binding with accoring field in the model
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/ConsentForm.controller.js:123]: Done

File: app/dteconsentappclient/webapp/controller/ConsentForm.controller.js [hunk 4]
```diff
@@ -91,33 +103,65 @@ sap.ui.define([
...
+			 * @returns {Boolean}
+			 */
+			isValidEmail: function(oControl, sValue){
+
+				// If emailId is valid set the value state to "None"
+				if(emailRegex.test(sValue)) {
+					oControl.setValueState("None");
+					return true;
+				}else{
+					// If emailId is Invalid , set the value state to "Error" with an error message
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/ConsentForm.controller.js:161]: Done

File: app/dteconsentappclient/webapp/controller/Enrollment.controller.js [hunk 5]
```diff
@@ -378,32 +377,56 @@ sap.ui.define([
...
 						if (control instanceof sap.m.Input && !control.getId().includes("-popup-input") || control instanceof sap.m.ComboBox || control instanceof sap.m.MaskInput) {
-		
-								// Get the binding path from the control
-								const bindingPath = control.getBinding('value')?.getPath() || control.getBinding("selectedKey")?.getPath();
+							
+							const userInput = control.getValue()
+							
+							// Validates that all required fields are filled; if a field is empty, marks it with an error state to indicate validation failure.
+							if((!userInput || userInput?.trim() === "") && control?.mProperties['required']) {
+								if(isShowError){
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/Enrollment.controller.js:385]: Done

File: app/dteconsentappclient/webapp/controller/ConsentForm.controller.js [hunk 6]
```diff
@@ -276,13 +327,20 @@ sap.ui.define([
 			}
 		},
 
+		validate: function(){
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/ConsentForm.controller.js:330]: Done

File: app/dteconsentappclient/webapp/controller/Enrollment.controller.js [hunk 7]
```diff
@@ -378,32 +377,56 @@ sap.ui.define([
...
+				 * @returns {Boolean}
+				 */
+				isValidEmail: function(oControl, sValue){
+
+					// If emailId is valid set the value state to "None"
+					if(emailRegex.test(sValue)) {
+						oControl.setValueState("None");
+						return true;
+					}else{
+						// If emailId is Invalid , set the value state to "Error" with an error message
```
[the remark this answers is not in the document]
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/controller/Enrollment.controller.js:419]: Done

File: app/dteconsentappclient/webapp/utils/ConsentAddressSuggestion.js (hunk 1 above)
Comment (Kokila-Kaavian) [reply app/dteconsentappclient/webapp/utils/ConsentAddressSuggestion.js:16]: Done

**Review by Divya-Kaavian — APPROVED** [review]

**Pull request merged**